From 95c2919e824254299494c46242ddfdf58bb4067f Mon Sep 17 00:00:00 2001
Message-ID: <95c2919e824254299494c46242ddfdf58bb4067f.1735936141.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1735936141.git.sharlatanus@gmail.com>
References: <cover.1735936141.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 3 Jan 2025 19:58:58 +0000
Subject: [PATCH 3/6] gnu: xtl: Improve package style.

* gnu/packages/cpp.scm (xtl): Shift order of fields.
[source] <url>: Fix redirection issue,
<https://github.com/QuantStack/xtl> permamently redirects to
<https://github.com/xtensor-stack/xtl>.
[arguments] <test-target>: Set to "xtest".
<phases>: Use default 'check.
[home-page]: Fix it.

Change-Id: Ib3bc50b7a9e798be8888c2b1eac832d6a524bdae
---
 gnu/packages/cpp.scm | 39 +++++++++++++++++----------------------
 1 file changed, 17 insertions(+), 22 deletions(-)

diff --git a/gnu/packages/cpp.scm b/gnu/packages/cpp.scm
index 151b01f7e5..121cc64f17 100644
--- a/gnu/packages/cpp.scm
+++ b/gnu/packages/cpp.scm
@@ -1061,30 +1061,25 @@ (define-public xtl
   (package
     (name "xtl")
     (version "0.7.5")
-    (source (origin
-              (method git-fetch)
-              (uri
-               (git-reference
-                (url "https://github.com/QuantStack/xtl")
-                (commit version)))
-              (sha256
-               (base32
-                "1llfy6pkzqx2va74h9xafjylyvw6839a843mqc05n6x6wll5bkam"))
-              (file-name (git-file-name name version))))
-    (native-inputs
-     (list doctest googletest nlohmann-json))
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/xtensor-stack/xtl")
+             (commit version)))
+       (sha256
+        (base32 "1llfy6pkzqx2va74h9xafjylyvw6839a843mqc05n6x6wll5bkam"))
+       (file-name (git-file-name name version))))
+    (build-system cmake-build-system)
     (arguments
      (list
-      #:configure-flags
-      #~(list "-DBUILD_TESTS=ON")
-      #:phases
-      #~(modify-phases %standard-phases
-          (replace 'check
-            (lambda _
-              (with-directory-excursion "test"
-                (invoke "./test_xtl")))))))
-    (home-page "https://github.com/QuantStack/xtl")
-    (build-system cmake-build-system)
+      #:configure-flags #~(list "-DBUILD_TESTS=ON")
+      #:test-target "xtest"))
+    (native-inputs
+     (list doctest
+           googletest
+           nlohmann-json))
+    (home-page "https://github.com/xtensor-stack/xtl")
     (synopsis "C++ template library providing some basic tools")
     (description "xtl is a C++ header-only template library providing basic
 tools (containers, algorithms) used by other QuantStack packages.")
-- 
2.46.0

